src/TreeViewList/TreeViewList.tsx

```diff
@@ -1,335 +1,216 @@
-
   return (
-    <TreeView
-      defaultCollapseIcon={<RemoveIcon />}
-      defaultExpandIcon={<AddIcon />}
-      expanded={expanded}
-      selected={selected}
-      onNodeToggle={handleToggle}
-      sx={{ width }}
-    >
-      {renderTree(parameters)}
-    </TreeView>
+    <Box sx={{ overflow: "clip", width }}>
+      {enableSearch ? (
+        <SearchBar
+          value={searchValue}
+          onChange={event => setSearchValue(event.target.value)}
+        />
+      ) : null}
+      <TreeView
+        key={`${searchValue} + ${defaultExpanded.length}`} // key to force re-render so that we can reset the expanded nodes when the search input changes but still allow user to expand/collapse nodes
+        defaultCollapseIcon={<RemoveIcon />}
+        defaultExpandIcon={<AddIcon />}
+        defaultExpanded={defaultExpanded}
+        selected={selected}
+        onNodeSelect={onNodeSelect}
+        onNodeToggle={onNodeToggle}
+      >
+        {renderTree(treeDisplayItems)}
+      </TreeView>
+    </Box>
   );
 };
 
-export default TreeViewList;
-
 /**
- * Creates a new tree node with the given properties.
+ * Creates a tree view list item with an optional tooltip.
  *
- * @param id - The ID of the new node.
- * @param name - The name of the new node.
- * @param tooltip - The tooltip of the new node.
- * @param [disable=true] - Optional parameter that indicates whether the node is disabled. Defaults to true.
- * @returns The newly created tree node.
+ * @param props - The properties for the tooltip tree item.
+ * @property props.disabled - If true, the tree item is disabled.
+ * @property props.label - The label of the tree item.
+ * @property props.nodeId - The unique ID of the tree item.
+ * @property props.tooltip - The tooltip to display.
+ * @returns The tree item wrapped in a tooltip.
  */
-function createNode(
-  id: string,
-  name: string,
-  tooltip?: string,
-  disable = true
-): TreeNode {
-  return {
-    children: [],
-    disable,
-    id,
-    name,
-    tooltip
-  };
-}
+const TooltipTreeItem = (
+  props: Pick<TreeNodeItem, "disabled" | "label" | "nodeId" | "tooltip"> & {
+    children?: React.ReactNode;
+  }
+) => {
+  return (
+    <Tooltip
+      disableFocusListener
+      title={props.tooltip ? <>{props.tooltip}</> : ""}
+      placement="right"
+    >
+      <TreeItem
+        {...props}
+        sx={theme => ({
+          color: theme.palette.text.primary,
+          padding: "5px"
+        })}
+      />
+    </Tooltip>
+  );
+};
 
-/**
- * Builds a tree structure from a flat list of items.
- *
- * @template T The type of the options array elements.
- * @param data - The flat list of items to transform into a tree.
- * @returns The tree structure built from the input items.
- * @example buildTree(items) // returns TreeNode[]
- */
-function buildTree<T>(data: Item<T>[]): TreeNode[] {
-  const itemsTree: TreeNode[] = [];
+// function to filter the items to contain only those that match the search term
```

Comment:
just expand the docs here to describe what the params are
```suggestion
/**
 * Filter Items to contain only those that match search
 * @param blah blah
 */
```

Reply:
Since this is a suggested code change it's best to have all the changes you suggest in the code, since you put 'blah blah' I can't commit this 😆